samples/excel/85-preview-apis/data-types-entity-attribution.yaml

```diff
@@ -0,0 +1,235 @@
+            const dataRange = productsTable.getDataBodyRange();
+            dataRange.load("values");
+
+            await context.sync();
+
+            // Set up the entities by mapping the product names to
+            // the sample JSON product data.
+            const entities = dataRange.values.map((rowValues) => {
+              // Get products and product properties.
+              const product = getProduct(rowValues[1]);
+
+              // Create entities by combining Products Table data and JSON data.
+              return [makeProductEntity(rowValues[1], rowValues[2], product)];
+            });
+
+            // Add the complete entities to the Products Table.
+            productColumn.getDataBodyRange().valuesAsJson = entities;
+
+            productColumn.getRange().format.autofitColumns();
+            await context.sync();
+          });
+        }
+
+        // Create entities with card layout and data provider fields. 
+        function makeProductEntity(productID: number, productName: string, product?: any) {
+          const entity: Excel.EntityCellValue = {
+            type: Excel.CellValueType.entity,
+            text: productName,
+            properties: {
+              "Product ID": {
+                type: Excel.CellValueType.string,
+                basicValue: productID.toString() || ""
+              },
+              "Product Name": {
+                type: Excel.CellValueType.string,
+                basicValue: productName || ""
+              },
+              "Quantity Per Unit": {
+                type: Excel.CellValueType.string,
+                basicValue: product.quantityPerUnit || ""
+              },
+              // Add Unit Price as a formatted number.
+              "Unit Price": {
+                type: Excel.CellValueType.formattedNumber,
+                basicValue: product.unitPrice,
+                numberFormat: "$* #,##0.00"
+              }
+            },
+            layouts: {
+              card: {
+                title: { property: "Product Name" },
+                sections: [
+                  {
+                    layout: "List",
+                    properties: ["Product ID"]
+                  },
+                  {
+                    layout: "List",
+                    title: "Quantity and price",
+                    collapsible: true,
+                    collapsed: false,
+                    properties: ["Quantity Per Unit", "Unit Price"]
+                  }
+                ]
+              }
+            },
+            provider: {
+              description: product.providerName, // Name of the data provider. Displays as a tooltip when hovering over the logo. Also displays as a fallback if the source address for the image is broken.
+              logoSourceAddress: product.sourceAddress, // Source URL of the logo to display.
+              logoTargetAddress: product.targetAddress // Destination URL that the logo navigates to when clicked.
```

Comment:
Since this block is what the sample is highlighting, is it possible to add this part as a separate button press? So, the workflow would be: 
- Add the table
- Add the entity values
- Add the provider data

That way, the difference between the two states would be more obvious.

Reply:
Based on offline conversation, I'll leave the workflow as is, and I added a sentence about the "data aggregator" scenario in the HTML.